triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/TriggerServiceFixture.scala

```diff
@@ -43,24 +55,60 @@ object TriggerServiceFixture {
       mat: Materializer,
       aesf: ExecutionSequencerFactory,
       ec: ExecutionContext): Future[A] = {
+    // Launch a toxiproxy instance . Wait on it to be ready to accept
+    // connections.
+    val toxiExe = BazelRunfiles.rlocation("external/toxiproxy_nix/bin/toxiproxy-cmd")
+    val toxiProxyPort = findFreePort()
+    val toxiCmd = toxiExe + " --port " + toxiProxyPort.toString
+    val toxiProxyProc = Process(toxiCmd).run()
+    var retries = 0
+    var scanning = true
+    while (scanning) {
+      try {
+        println("Waiting for Toxiproxy...")
+        val channel = new java.net.Socket("localhost", toxiProxyPort)
```

Comment:
In general, I'd advise preferring an `InetAddress` instance over a string here. You can use `InetAddress.getLoopbackAddress`. `Socket` will do the conversion from `String` to `InetAddress` for you, but it's an unnecessary jump.

`Socket` will accept one of these but unfortunately, it seems Toxiproxy likes strings. If you want to stay a little more type-safe and coerce to string at the last moment, you can store `val host = InetAddress.getLoopbackAddress` and call `getHostName` when you need it.